Automattic/jetpack
# Fix calculated price for Jetpack Search in product description

## Summary

The current product description at `/wp-admin/admin.php?page=jetpack#/product/search` doesn't show the correct price. It doesn't account for sale coupons, and the search product's intro discount is done using a sale coupon. Additionally, we would like this page to recommend a JP Search tier based on how many records the site has.

#### Changes proposed in this Pull Request:

The product description component needs to consider sale coupons when calculated discounted price. Not only because the Search intro discount solely relies on it, but also because we want sales to be represented here when all products are on an additional sale for some holiday.

The product description specifically for JP search should take into account how many records the current site has, and recommend a plan based on that.

A data query component was created to query the `/jetpack/v4/site/products` endpoint. Previously, the products for these Product Descriptions were pulling their data from `/jetpack/v4/products` endpoint. The former seems to return correct pricing for the JP Search plans, taking into account post count on the site, where the latter did not. 

#### Jetpack product discussion
This came from [this discussion](https://github.com/Automattic/jetpack/pull/24461#issuecomment-1137768582)

#### Does this pull request change what data or activity we track or use?

My PR does not change any data tracking. I am using a different API endpoint to pull products into the Product Description, but this came from an existing Redux action, so I am not sure if this needs to be looked at by somebody or not

#### Testing instructions:

1. Start your local environment and start on branch `trunk`
2. Go to `/wp-admin/admin.php?page=jetpack#/product/search`, you will need to have at least Jetpack Free setup on your local environment and you cannot be in "offline mode." (I had a problem with being in offline mode when using localhost, so I used Jurassic Tube for this)
3. Confirm that the Search price on this page does not include the sale coupon intro discount, and that it does not change based on the number of records on your site
![image](https://user-images.githubusercontent.com/65001528/176928806-6105ab8e-daca-4727-b624-d0f2277c2997.png)
   - To test out the number of records, I imported records from [these files](https://github.com/WPTT/theme-test-data). This is only 50ish posts, so to get 100 I changed the titles and post-id's to all the posts via search and replace to make them unique again (if there is a better way to do this, please let me know 😄 
4. Switch to this branch: `fix/search-price-on-wp-product-page`
5. Refresh the page (make sure you have `jetpack watch plugins/jetpack` running to see the changes)
6. Confirm that the intro discount is now shown on this page and that the price changes depending on the number of posts on the site
![image](https://user-images.githubusercontent.com/65001528/176929345-83a9953e-eded-4b0c-956e-a9226dad3425.png)

Note: When importing posts, the site takes at least a few minutes to be re-indexed and recognize the new posts, so you may have to wait a bit to see the next tier of search pricing on this page after importing the content.
